TicketsDemo.Domain/DefaultImplementations/TicketServiceLoggingDecorator.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Domain.Interfaces;
+
+namespace TicketsDemo.Domain.DefaultImplementations
+{
+    abstract class TicketServiceLoggingDecorator : ITicketService
+    {
+        private ITicketService _decoratedobject;
+        string choice = "F";
```

Comment:
having "choise" here cause couple of errors.

1 - three responsibilities are merged. 1 - "what are we going to log", 2 - "how we are logging to debug window", 3 - "how we are logging to file". (which stands against the single responsibility and open-closed principles).

Create separate ILogger interface (or reuse existing one). Then create two different implementations - DebugLogger and FileLogger. 

ILogger to be injected into the TicketServiceLoggingDecorator using the dependency injection.
Proper implementation to be registered in the NinjectWebCommon